otj-kafka/src/main/java/com/opentable/kafka/builders/KafkaBaseBuilder.java

```diff
@@ -111,68 +107,113 @@ void setupBootstrap() {
         return root;
     }
 
-    void removeProperty(String key) {
+    public SELF removeProperty(String key) {
         finalProperties.remove(key);
+        return self();
+    }
+
+    public SELF withProperty(String key, Object value) {
+        this.addProperty(key, value);
+        return self();
+    }
+
+
+    public SELF withProperties(final Map<String, Object> config) {
+        this.addProperties(config);
+        return self();
     }
 
-    void withPrefix(String metricsPrefix) {
+    protected SELF withPrefix(String metricsPrefix) {
         if ((metricsPrefix != null) && (!metricsPrefix.startsWith(OtMetricsReporterConfig.DEFAULT_PREFIX + "."))) {
                 metricsPrefix = OtMetricsReporterConfig.DEFAULT_PREFIX + "." + metricsPrefix;
         }
         this.metricsPrefix = Optional.ofNullable(metricsPrefix);
+        return self();
     }
 
-    void withBootstrapServer(String bootStrapServer) {
+    public SELF withBootstrapServer(String bootStrapServer) {
         this.bootStrapServers.add(bootStrapServer);
+        return self();
     }
 
-    void withBootstrapServers(List<String> bootStrapServers) {
+    public SELF withBootstrapServers(List<String> bootStrapServers) {
         this.bootStrapServers.addAll(bootStrapServers);
+        return self();
     }
 
-    void withClientId(String val) {
+    public SELF withClientId(String val) {
         clientId = Optional.ofNullable(val);
+        return self();
     }
 
-    void withSecurityProtocol(SecurityProtocol protocol) {
+    public SELF withSecurityProtocol(SecurityProtocol protocol) {
         this.securityProtocol = Optional.ofNullable(protocol);
+        return self();
     }
 
-    void withRequestTimeoutMs(Duration duration) {
+    public SELF withRequestTimeoutMs(Duration duration) {
         requestTimeout = OptionalLong.of(duration.toMillis());
+        return self();
     }
-    void withRetryBackOff(Duration duration) {
+
+    public SELF withRequestTimeout(Duration duration) {
+        if (duration != null) {
+            withRequestTimeoutMs(duration);
+        }
+        return self();
+    }
+
+    public SELF withRetryBackOff(Duration duration) {
         retryBackoff = OptionalLong.of(duration.toMillis());
+        return self();
     }
 
-    void withMetricRegistry(MetricRegistry metricRegistry) {
+    public SELF withRetryBackoff(Duration duration) {
+        if (duration != null) {
+            withRetryBackOff(duration);
+        }
+        return self();
+    }
+
+    public SELF withMetricRegistry(MetricRegistry metricRegistry) {
         this.metricRegistry = Optional.ofNullable(metricRegistry);
+        return self();
     }
 
-    void withMetrics(final boolean enabled) {
+    public SELF withMetricRegistry(MetricRegistry metricRegistry, String metricsPrefix) {
+        withMetricRegistry(metricRegistry);
+        withPrefix(metricsPrefix);
+        return self();
+    }
+
+    public SELF withMetrics(final boolean enabled) {
```

Comment:
this is confusing to expose. In general consider the issue of exposure to the PUBLIC